tests/postgres/select_test.go

```diff
@@ -2251,6 +2251,37 @@ FOR`
 	}
 }
 
+func TestRowLockWithJoins(t *testing.T) {
+	expectedSQL := `
+SELECT *
+FROM dvds.film
+     INNER JOIN dvds.film_category ON (film_category.film_id = film.film_id)
+     LEFT JOIN dvds.film_actor ON (film_actor.film_id = film.film_id)
+LIMIT 1
+FOR UPDATE OF "film", "film_category" NOWAIT;
+`
+	query := Film.
+		SELECT(STAR).
+		FROM(
+			Film.
+				INNER_JOIN(FilmCategory, FilmCategory.FilmID.EQ(Film.FilmID)).
+				LEFT_JOIN(FilmActor, FilmActor.FilmID.EQ(Film.FilmID))).
+		LIMIT(1).
+		FOR(UPDATE().OF(Film, FilmCategory).NOWAIT())
+
+	testutils.AssertDebugStatementSql(t, query, expectedSQL, int64(1))
+
+	tx, _ := db.Begin()
+
+	res, err := query.Exec(tx)
+	require.NoError(t, err)
+	rowsAffected, _ := res.RowsAffected()
+	require.Equal(t, rowsAffected, int64(1))
+
+	err = tx.Rollback()
+	require.NoError(t, err)
```

Comment:
2274-2282, can be replaced with a `testutils.AssertExecAndRollback(t, query, db, 1)`